### PR TITLE
groups: check permissions

### LINKED ISSUE #17: System/managed groups + flexible permissions
The current permission model does not allow the following at the same time:
 - a user can view a list all of here groups
 - a user cannot leave specific groups (system/managed groups)
 - a user can join+leave specific groups (non-system/unmanaged groups)
 - (group) admins still can do whatever they want OR admins have to follow the same regulations when adding/removing users to groups (*to decide*)

Is the above use case what `is_managed` was intended to be for?

### PR DESCRIPTION
**Pull request by crepererum**
* Reintegrates permission checks after module rewrite.
  (closes #17)

Signed-off-by: Marco Neumann <marco@crepererum.net>

Can someone please have a closer look on that, because it affects security? CC @lnielsen

### CODE REVIEW

File: invenio_groups/views/settings.py [hunk 1]
```diff
@@ -188,12 +208,12 @@ def manage(group_id):
                 form=form,
                 group=group,
             )
-
-    return render_template(
-        "groups/new.html",
-        form=form,
-        group=group,
-    )
+    else:
```
Comment (jirikuncar) [inline invenio_groups/views/settings.py]: IMHO the else should not be here. Can you verify it?

File: invenio_groups/views/settings.py [hunk 2]
```diff
@@ -230,23 +253,33 @@ def delete(group_id):
...
+        return render_template(
+            "groups/members.html",
+            group=group,
+            members=members,
+            page=page,
+            per_page=per_page,
+            q=q,
+            s=s,
+        )
+    else:
```
Comment (jirikuncar) [inline invenio_groups/views/settings.py]: Personally I would prefer no `else` branch if you've previously finished with `return`.

**Comment by jirikuncar** [comment]
@crepererum can you also extend our test cases?

**Comment by crepererum** [comment]
@jirikuncar testing the views, without depending on Invenio (including: the washers, the forms, the db object, etc.)?

**Comment by jirikuncar** [comment]
For the time being we can install it on Travis (see #20).

**Comment by lnielsen** [comment]
Commit message: deposit -> groups?

**Comment by crepererum** [comment]
@lnielsen you are right, my fault.

**Comment by crepererum** [comment]
New version, tests are still missing.

File: invenio_groups/models.py [hunk 3]
```diff
@@ -468,13 +468,62 @@ def can_see_members(self, user):
...
+        else:
+            return False
+
+    def can_leave(self, user):
+        """Determine if user can leave a group.
+
+        :param user: User to be checked.
+        :returns: True or False.
+        """
+        if hasattr(user, 'is_superadmin') and user.is_superadmin:
```
Comment (jirikuncar) [inline invenio_groups/models.py]: WDYT about `getattr(user, 'is_superadmin')`?

File: invenio_groups/views/settings.py [hunk 4]
```diff
@@ -201,15 +213,25 @@ def manage(group_id):
...
+        try:
+            group.delete()
+        except Exception as e:
+            flash(str(e), "error")
+            return redirect(url_for(".index"))
+
+        flash(_('Successfully removed group "%(group_name)s"',
+                group_name=group.name), 'success')
+
+    flash(
```
Comment (jirikuncar) [inline invenio_groups/views/settings.py]: there is something weird with the flow and flash messages. can you verify it?

**Pull request closed without merging**